tests/CMakeLists.txt

```diff
@@ -1 +1,56 @@
+# Add an executable and its related test.
+#
+# The executable is always linked to 'kokkostools' and 'test_common'.
+#
+# The test environment always contains 'KOKKOS_TOOLS_LIBS' that is set as the path
+# to the target file of 'CONNECTOR_LIB'.
+#
+# Arguments:
+#   TARGET_NAME   : name of the test
+#   SOURCE_FILE   : source file, defaults to <TARGET_NAME>.cpp
+#   CONNECTOR_LIB : connector library target name under test
+function(kp_add_executable_and_test)
+
+    cmake_parse_arguments(kaeat_args "" "TARGET_NAME;SOURCE_FILE;CONNECTOR_LIB" "" ${ARGN})
+
+    if(NOT DEFINED kaeat_args_TARGET_NAME OR NOT DEFINED kaeat_args_SOURCE_FILE OR NOT DEFINED kaeat_args_CONNECTOR_LIB)
+        message(FATAL_ERROR "Wrong usage.")
+    endif()
+
+    add_executable(${kaeat_args_TARGET_NAME})
+
+    target_sources(
+        ${kaeat_args_TARGET_NAME}
+        PRIVATE
+            ${kaeat_args_SOURCE_FILE}
+    )
+    target_link_libraries(
+        ${kaeat_args_TARGET_NAME}
+        PRIVATE
+            kokkostools test_common
+    )
+
+    add_test(
+        NAME ${kaeat_args_TARGET_NAME}
+        COMMAND $<TARGET_FILE:${kaeat_args_TARGET_NAME}>
+    )
+    set_property(
+        TEST ${kaeat_args_TARGET_NAME}
+        APPEND
+        PROPERTY
+            ENVIRONMENT "KOKKOS_TOOLS_LIBS=$<TARGET_FILE:${kaeat_args_CONNECTOR_LIB}>"
+    )
```

Comment:
Why is it not guarded with `if(kaeat_args_CONNECTOR_LIB)`?
Is there a good reason to make it required to pass that argument?

Reply:
I think that a lot of (not-yet-written) tests will need this environment variable... But I've made it optional :ballot_box_with_check: